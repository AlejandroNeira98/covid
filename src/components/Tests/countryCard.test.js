import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router} from 'react-router-dom';
import CountryCard from '../countryCard';
import store from '../../redux/configureStore';

it('works', () => {
  const tree = render(
    <Provider store={store}>
      <Router>
        <CountryCard />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  render(
    <Provider store={store}>
      <Router>
        <CountryCard />
      </Router>
    </Provider>,
  );
  expect(screen.getByText(/Cases Confirmed:/)).toBeInTheDocument();
});
