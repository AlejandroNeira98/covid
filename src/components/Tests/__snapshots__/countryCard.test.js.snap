// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div>
          <div
            class="name"
          />
          <div>
            Last 24h update:
          </div>
        </div>
        <div>
          <ul>
            <li>
              <span>
                 Cases Confirmed:
              </span>
              <span />
            </li>
            <li>
              <span>
                New Recovered:
              </span>
              <span />
            </li>
            <li>
              <span>
                New Deaths:
              </span>
              <span>
                 
                 
              </span>
            </li>
          </ul>
        </div>
        <div
          class="button"
        >
          <a
            class=""
            href="/Country/undefined/undefined"
          >
            <button
              class="button"
              type="button"
            >
              See more
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div>
        <div
          class="name"
        />
        <div>
          Last 24h update:
        </div>
      </div>
      <div>
        <ul>
          <li>
            <span>
               Cases Confirmed:
            </span>
            <span />
          </li>
          <li>
            <span>
              New Recovered:
            </span>
            <span />
          </li>
          <li>
            <span>
              New Deaths:
            </span>
            <span>
               
               
            </span>
          </li>
        </ul>
      </div>
      <div
        class="button"
      >
        <a
          class=""
          href="/Country/undefined/undefined"
        >
          <button
            class="button"
            type="button"
          >
            See more
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div>
          <div
            class="name"
          />
          <div>
            Last 24h update:
          </div>
        </div>
        <div>
          <ul>
            <li>
              <span>
                 Cases Confirmed:
              </span>
              <span />
            </li>
            <li>
              <span>
                New Recovered:
              </span>
              <span />
            </li>
            <li>
              <span>
                New Deaths:
              </span>
              <span>
                 
                 
              </span>
            </li>
          </ul>
        </div>
        <div
          class="button"
        >
          <a
            class=""
            href="/Country/undefined/undefined"
          >
            <button
              class="button"
              type="button"
            >
              See more
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div>
        <div
          class="name"
        />
        <div>
          Last 24h update:
        </div>
      </div>
      <div>
        <ul>
          <li>
            <span>
               Cases Confirmed:
            </span>
            <span />
          </li>
          <li>
            <span>
              New Recovered:
            </span>
            <span />
          </li>
          <li>
            <span>
              New Deaths:
            </span>
            <span>
               
               
            </span>
          </li>
        </ul>
      </div>
      <div
        class="button"
      >
        <a
          class=""
          href="/Country/undefined/undefined"
        >
          <button
            class="button"
            type="button"
          >
            See more
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
